.github/workflows/test.yaml

```diff
@@ -339,6 +342,8 @@ jobs:
   # process late in the release cycle,
   # The files are published only when a tag is created.
   release-publish:
+    permissions:
+      contents: write
```

Comment:
ok. so we need this in write mode, only to update the "stable" branch.

The "stable" branch was added as a hack, since Read the docs, doesn't recognizes our `twisted-12.23.45` tag format, and can't determine the latest stable version.

Maybe for better security, the "stable" branch update can be moved into a separate job that is performed after release-publish.

So, this would go further with the principle of least privilege

Reply:
Yes, you got the idea! Restrict as much as possible. I just usually try to minimize the changes to avoid breaking anything. I have split it into a separate job. Please, take a look.